great_expectations/dataset/sqlalchemy_dataset.py

```diff
@@ -511,7 +513,9 @@ def __init__(
             # NOTE: Eugene 2020-01-31: @James, this is a not a proper fix, but without it the "public" schema
             # was used for a temp table and raising an error
             schema = None
-            table_name = generate_temporary_table_name()
+            table_name = generate_temporary_table_name(
+                default_table_name_prefix="#ge_temp_"
```

Comment:
I don't believe `#` is needed here because it is added below if dialect is mssql.

Reply:
Fixed.